### ISSUE TITLE
Conflicting declarations with os-install

### ISSUE THREAD

**Issue by aither64** [vpsfreecz/vpsadminos#72]
Ah sorry, I misunderstood. If you prefer, I can make a new ticket instead
of resurrecting this one.

This is after running `os-install --root /mnt`.

The conflicting declarations are in:
- /root/.nix-defexpr/channels/vpsadminos/os/modules/system/activation/top-level.nix
- /root/.nix-defexpr/channels/vpsadminos/os/modules/system/boot/loader/grub/grub.nix

I did a `nix-channel --update` to get the installer to work in the first
place. Vpsadminos channel is tracking staging.

I am building derivationn vpsadminos-24.05.0pre-git

_Originally posted by @Zurga in https://github.com/vpsfreecz/vpsadminos/issues/56#issuecomment-2329655448_

**Comment by aither64** [comment]
Ok, I'll try to look into it. Something is most likely outdated and out-of-sync with nixpkgs. Honestly, while the ISO is the easiest way to get into vpsAdminOS and that is why I'm making them, it doesn't get any testing at all. There's only two of us working on this project and what we don't directly use ourselves doesn't get much attention. That includes on-disk installation, we only use vpsAdminOS as a live system. So there might be a lot of rough edges :)

**Comment by aither64** [comment]
I've pushed fixes to staging that make it possible to install vpsAdminOS on disk. The conflicting declaration error can be fixed by adding `boot.isLiveSystem = false;` to your configuration (`os-generate-config` now includes it). Also make sure that initrd has modules necessary to mount the root filesystem. I used ext4 and `boot.initrd.supportedFilesystems.ext4 = true;` wasn't enough, I had to set `boot.initrd.kernelModules = [ "ext4" ];`. I think NixOS handles it better or something changed there, I haven't investigated it.

I'd say that you could update the channel and try it, but channel updates are broken for me (some conflict with artwork/boot.png). I could rebuild the ISO available at iso.vpsadminos.org, although without channel updates it is still broken, so more work is needed.

**Comment by aither64** [comment]
Channel updates are now fixed as well. `nix-channel --update` will work even from older systems, the problem was only with updating individual channels (either nixos or vpsadminos). I've also deployed an up-to-date ISO to https://iso.vpsadminos.org, that will save you from building the kernel yourself.

`os-generate-config` and `os-install` should now make it possible to install vpsAdminOS on disk. There are definitely many things that could be further polished, but at least the tools aren't completely broken anymore.
